Jint/Native/Math/MathInstance.cs

```diff
@@ -650,6 +653,89 @@ private static JsValue Tan(JsValue thisObject, JsValue[] arguments)
             return System.Math.Tan(x);
         }
 
+        private static JsValue Truncate(JsValue thisObject, JsValue[] arguments)
+        {
+            var x = TypeConverter.ToNumber(arguments.At(0));
+
+            if (double.IsNaN(x))
+            {
+                return double.NaN;
+            }
+
+            if (NumberInstance.IsPositiveZero(x) || NumberInstance.IsNegativeZero(x))
+            {
+                return x;
+            }
+
+            if (double.IsPositiveInfinity(x))
+            {
+                return double.PositiveInfinity;
+            }
+
+            if (double.IsNegativeInfinity(x))
+            {
+                return double.NegativeInfinity;
+            }
+
+            return System.Math.Truncate(x);
+        }
+
+        private static JsValue Sign(JsValue thisObject, JsValue[] arguments)
+        {
+            var x = TypeConverter.ToNumber(arguments.At(0));
+
+            if (double.IsNaN(x))
+            {
+                return double.NaN;
+            }
+
+            if (NumberInstance.IsPositiveZero(x) || NumberInstance.IsNegativeZero(x))
+            {
+                return x;
+            }
+
+            if (double.IsPositiveInfinity(x))
+            {
+                return 1;
+            }
+
+            if (double.IsNegativeInfinity(x))
+            {
+                return -1;
+            }
+
+            return System.Math.Sign(x);
+        }
+
+        private static JsValue Cbrt(JsValue thisObject, JsValue[] arguments)
+        {
+            var x = TypeConverter.ToNumber(arguments.At(0));
 
+            if (double.IsNaN(x))
+            {
+                return double.NaN;
+            }
+            else if (NumberInstance.IsPositiveZero(x))
+            {
+                return +0;
+            }
+            else if (NumberInstance.IsNegativeZero(x))
+            {
+                return -0;
+            }
+            else if (double.IsPositiveInfinity(x))
+            {
+                return double.PositiveInfinity;
```

Comment:
Don't we have constants for these?

Reply:
We don't have a ECMA test for this, but I tweaked the code to follow same pattern as the two above to return value as-is if it's either positive or negative infinity.